app/models/upload.rb

```diff
@@ -392,30 +399,38 @@ def short_url_basename
 #
 # Table name: uploads
 #
-#  id                :integer          not null, primary key
-#  user_id           :integer          not null
-#  original_filename :string           not null
-#  filesize          :integer          not null
-#  width             :integer
-#  height            :integer
-#  url               :string           not null
-#  created_at        :datetime         not null
-#  updated_at        :datetime         not null
-#  sha1              :string(40)
-#  origin            :string(1000)
-#  retain_hours      :integer
-#  extension         :string(10)
-#  thumbnail_width   :integer
-#  thumbnail_height  :integer
-#  etag              :string
-#  secure            :boolean          default(FALSE), not null
+#  id                     :integer          not null, primary key
+#  user_id                :integer          not null
+#  original_filename      :string           not null
+#  filesize               :integer          not null
+#  width                  :integer
+#  height                 :integer
+#  url                    :string           not null
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#  sha1                   :string(40)
+#  origin                 :string(1000)
+#  retain_hours           :integer
+#  extension              :string(10)
+#  thumbnail_width        :integer
+#  thumbnail_height       :integer
+#  etag                   :string
+#  secure                 :boolean          default(FALSE), not null
```

Comment:
do we need the "secure" column give we have "access_control_post_id" ?

Reply:
Yes, because we don't know the `access_control_post_id` when the upload is first created but we still want to know it was created in a secure context, as the ACL is now set at upload time (and the URL for /secure-media-uploads/ is used straight away for posts).

Reply:
aha yeah ... makes sense